### PR TITLE
Cleanup settings Application class

### PR DESCRIPTION
**Pull request by nickvergessen**
@MorrisJobke @LukasReschke @rullzer

### CODE REVIEW

**Comment by LukasReschke** [comment]
```
1) Tests\Settings\Controller\GroupsControllerTest::testIndexSortByName
627s
101
InvalidArgumentException: Identifier "GroupsController" is not defined.
627s
103
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
104
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/GroupsControllerTest.php:48
627s
106
2) Tests\Settings\Controller\GroupsControllerTest::testIndexSortbyCount
627s
107
InvalidArgumentException: Identifier "GroupsController" is not defined.
627s
109
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
110
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/GroupsControllerTest.php:48
627s
112
3) Tests\Settings\Controller\GroupsControllerTest::testCreateWithExistingGroup
627s
113
InvalidArgumentException: Identifier "GroupsController" is not defined.
627s
115
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
116
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/GroupsControllerTest.php:48
627s
118
4) Tests\Settings\Controller\GroupsControllerTest::testCreateSuccessful
627s
119
InvalidArgumentException: Identifier "GroupsController" is not defined.
627s
121
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
122
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/GroupsControllerTest.php:48
627s
124
5) Tests\Settings\Controller\GroupsControllerTest::testCreateUnsuccessful
627s
125
InvalidArgumentException: Identifier "GroupsController" is not defined.
627s
127
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
128
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/GroupsControllerTest.php:48
627s
130
6) Tests\Settings\Controller\GroupsControllerTest::testDestroySuccessful
627s
131
InvalidArgumentException: Identifier "GroupsController" is not defined.
627s
133
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
134
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/GroupsControllerTest.php:48
627s
136
7) Tests\Settings\Controller\GroupsControllerTest::testDestroyUnsuccessful
627s
137
InvalidArgumentException: Identifier "GroupsController" is not defined.
627s
139
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
140
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/GroupsControllerTest.php:48
627s
142
8) Tests\Settings\Controller\LogSettingsControllerTest::testSetLogLevel with data set #0 (-1, false)
627s
143
InvalidArgumentException: Identifier "LogSettingsController" is not defined.
627s
145
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
146
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/LogSettingsControllerTest.php:34
627s
148
9) Tests\Settings\Controller\LogSettingsControllerTest::testSetLogLevel with data set #1 (0, true)
627s
149
InvalidArgumentException: Identifier "LogSettingsController" is not defined.
627s
151
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
152
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/LogSettingsControllerTest.php:34
627s
154
10) Tests\Settings\Controller\LogSettingsControllerTest::testSetLogLevel with data set #2 (1, true)
627s
155
InvalidArgumentException: Identifier "LogSettingsController" is not defined.
627s
157
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
158
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/LogSettingsControllerTest.php:34
627s
160
11) Tests\Settings\Controller\LogSettingsControllerTest::testSetLogLevel with data set #3 (2, true)
627s
161
InvalidArgumentException: Identifier "LogSettingsController" is not defined.
627s
163
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
164
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/LogSettingsControllerTest.php:34
627s
166
12) Tests\Settings\Controller\LogSettingsControllerTest::testSetLogLevel with data set #4 (3, true)
627s
167
InvalidArgumentException: Identifier "LogSettingsController" is not defined.
627s
169
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
170
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/LogSettingsControllerTest.php:34
627s
172
13) Tests\Settings\Controller\LogSettingsControllerTest::testSetLogLevel with data set #5 (4, true)
627s
173
InvalidArgumentException: Identifier "LogSettingsController" is not defined.
627s
175
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
176
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/LogSettingsControllerTest.php:34
627s
178
14) Tests\Settings\Controller\LogSettingsControllerTest::testSetLogLevel with data set #6 (5, false)
627s
179
InvalidArgumentException: Identifier "LogSettingsController" is not defined.
627s
181
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
182
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/LogSettingsControllerTest.php:34
627s
184
15) Tests\Settings\Controller\LogSettingsControllerTest::testDownload
627s
185
InvalidArgumentException: Identifier "LogSettingsController" is not defined.
627s
187
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
188
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/LogSettingsControllerTest.php:34
627s
190
16) Tests\Settings\Controller\MailSettingsControllerTest::testSetMailSettings
627s
191
InvalidArgumentException: Identifier "MailSettingsController" is not defined.
627s
193
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
194
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/MailSettingsControllerTest.php:109
627s
196
17) Tests\Settings\Controller\MailSettingsControllerTest::testStoreCredentials
627s
197
InvalidArgumentException: Identifier "MailSettingsController" is not defined.
627s
199
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
200
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/MailSettingsControllerTest.php:152
627s
202
18) Tests\Settings\Controller\MailSettingsControllerTest::testSendTestMail
627s
203
InvalidArgumentException: Identifier "MailSettingsController" is not defined.
627s
205
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
206
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/MailSettingsControllerTest.php:191
627s
208
19) Tests\Settings\Controller\SecuritySettingsControllerTest::testTrustedDomainsWithExistingValues
627s
209
InvalidArgumentException: Identifier "SecuritySettingsController" is not defined.
627s
211
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
212
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/SecuritySettingsControllerTest.php:32
627s
214
20) Tests\Settings\Controller\SecuritySettingsControllerTest::testTrustedDomainsEmpty
627s
215
InvalidArgumentException: Identifier "SecuritySettingsController" is not defined.
627s
217
/drone/src/github.com/nextcloud/server/3rdparty/pimple/pimple/src/Pimple/Container.php:96
627s
218
/drone/src/github.com/nextcloud/server/tests/Settings/Controller/SecuritySettingsControllerTest.php:32

```

File: tests/Settings/ApplicationTest.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+use OCP\AppFramework\Middleware;
+use OCP\IUser;
+use OCP\IUserSession;
+use Test\TestCase;
+
+/**
+ * Class ApplicationTest
+ *
+ * @package Tests\Settings
+ * @group DB
```
Comment (LukasReschke) [inline tests/Settings/ApplicationTest.php]: Just asking: Do they really require the DB? (didn't test locally, if a constructor does: No problem! :))

Comment (nickvergessen) [inline tests/Settings/ApplicationTest.php]: One service uses them yes, and I'm too lazy to split the test into two classes

Comment (rullzer) [inline tests/Settings/ApplicationTest.php]: Can't you tag independent tests?

**Comment by nickvergessen** [comment]
Updated the tests 🙈

File: tests/Settings/ApplicationTest.php (hunk 1 above)
Comment (nickvergessen) [inline tests/Settings/ApplicationTest.php]: They are all in one test, using a data provider... anyway not worth any effort from my pov

**Comment by MorrisJobke** [comment]
Code looks good, tested and works :+1:

**Review by MorrisJobke — APPROVED** [review]

**Comment by rullzer** [comment]
Whee less code!
LGTM

**Pull request merged**